cmake/External/aotriton.cmake

```diff
@@ -65,12 +140,19 @@ if(NOT __AOTRITON_INCLUDED)
       -DAOTRITON_GPU_BUILD_TIMEOUT=0
       -DAOTRITON_NO_PYTHON=ON
       -DAOTRITON_NOIMAGE_MODE=${noimage}
-      BUILD_BYPRODUCTS "${__AOTRITON_INSTALL_DIR}/lib/libaotriton_v2.so"
+      -DHIP_PLATFORM=amd
```

Comment:
I see that https://github.com/ROCm/pytorch/pull/2686 backported another aotriton change to the release/2.8 branch. Do we want at least this line changed for 2.8 too? We won't be able to build aotriton on Linux without it. I'm less concerned with backporting the full Windows support to 2.8.

Reply:
If you're referring to https://github.com/ROCm/pytorch/pull/2686/commits/a86374d594a831acad26751b47d8df4dea488d3e, yeah we should have that I think. Also https://github.com/ROCm/pytorch/pull/2686/commits/c497508b3a83a1a6293f7c5c842614eec613f2f7. The 2.9 branch cut was around september 9th, so any commits in that PR after sept 9th should count.

Reply:
@xinyazhang can you confirm if we need the above 2 commits for release/2.9 ?

Reply:
> @xinyazhang can you confirm if we need the above 2 commits for release/2.9 ?

If you need Windows support then you need https://github.com/ROCm/pytorch/commit/a86374d594a831acad26751b47d8df4dea488d3e . This PR is to fix bug introduced when adding Windows support.
https://github.com/ROCm/pytorch/commit/c497508b3a83a1a6293f7c5c842614eec613f2f7 is also needed , which fixes the `persistent_counter` usage.